packages/replay-next/components/inspector/PropertiesRenderer.tsx

```diff
@@ -137,6 +145,41 @@ export default function PropertiesRenderer({
     );
   }
 
+  let renderedFunctionLocation: ReactNode = null;
+  if (preview && object.className == "Function" && preview.functionLocation) {
+    const sourcesById = sourcesByIdCache.read(client);
+    const functionDefinition = preview.functionLocation;
+    const preferredFunctionLocation = getPreferredLocation(
+      sourcesById,
+      preferredGeneratedSourceIds,
+      functionDefinition
+    );
+    const source = sourcesById.get(preferredFunctionLocation.sourceId);
+    const fileName = source ? getSourceFileName(source) : preferredFunctionLocation.sourceId;
+    const fnLocation = `${fileName}:${preferredFunctionLocation.line}`;
+
+    renderedFunctionLocation = (
+      <KeyValueHeader
+        value={
+          <span
+            onClick={() => {
+              if (inspectFunctionDefinition !== null) {
+                inspectFunctionDefinition(functionDefinition);
+              }
+            }}
+            className={styles.SourceLocationLink}
+            data-test-name="ClientValue"
+          >
+            {fnLocation}
+          </span>
+        }
+        nameClass={styles.BucketLabel}
+        layout="vertical"
+        name="[[FunctionLocation]]"
+      />
+    );
+  }
+
```

Comment:
What if we did something close to what you're doing, but moving things around a bit?

1. Move the `renderedPrototype` into its own React component e.g. `PrototypeRenderer` (in `replay-next/components/inspector/properties`?) instead of that logic being inline in `PropertiesRenderer`
2. Add a `FunctionRenderer` component in that folder that does the location (and also includes/renders the prototype)

Then we could decide which one to render like so
```tsx
let specialPropertiesRenderer: ReactNode = null;
switch (className) {
  case "Function":
    specialPropertiesRenderer = <FunctionRenderer object={object} />;
    break;
  default:
    specialPropertiesRenderer = <PrototypeRenderer object={object} />;
    break;
}
```

I think we could avoid touching `KeyValueRenderer` at all this way, and keep type-specific logic out of `PropertiesRenderer`

Does that make sense?

Reply:
Yep, working on this